src/validation/validate.js

```diff
@@ -51,3 +51,39 @@ export function validate(
   visit(documentAST, visitWithTypeInfo(typeInfo, visitor));
   return context.getErrors();
 }
+
+export function validateSDL(
+  documentAST: DocumentNode,
+  schemaToExtend?: ?GraphQLSchema,
```

Comment:
I like this a lot, and would in fact be a fan of moving all existing schema validation rules into `validateSDL`. Not sure if we want to do that prior to `14.0`, but it might be nice to add them as optional rules for `14.1`, and then have those be the specified rules for `15.0` (6 months down the line? Not sure what our release cadence is supposed to be).

I am very tempted to fix `buildASTSchema(ast, {assumeValid: true})` to not throw on errors that would normally be considered SDL-type errors (i.e. errors that do not prevent us from building a schema, it just would likely fail once built).